ledger/eval_test.go

```diff
@@ -85,6 +95,52 @@ func TestBlockEvaluator(t *testing.T) {
 	err = eval.Transaction(st, transactions.ApplyData{})
 	require.NoError(t, err)
 
+	// Broken signature should fail
+	stbad := st
+	st.Sig[2] ^= 8
+	txgroup := []transactions.SignedTxn{stbad}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+
+	// Repeat should fail
+	txgroup = []transactions.SignedTxn{st}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+	err = eval.Transaction(st, transactions.ApplyData{})
+	require.Error(t, err)
+
+	// out of range should fail
+	btxn := txn
+	btxn.FirstValid++
+	btxn.LastValid += 2
+	st = btxn.Sign(keys[0])
+	txgroup = []transactions.SignedTxn{st}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+	err = eval.Transaction(st, transactions.ApplyData{})
+	require.Error(t, err)
+
+	// bogus group should fail
+	btxn = txn
+	btxn.Group[1] = 1
+	st = btxn.Sign(keys[0])
+	txgroup = []transactions.SignedTxn{st}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+	err = eval.Transaction(st, transactions.ApplyData{})
+	require.Error(t, err)
+
+	// mixed fields should fail
+	btxn = txn
+	btxn.XferAsset = 3
+	st = btxn.Sign(keys[0])
+	txgroup = []transactions.SignedTxn{st}
+	err = eval.TestTransactionGroup(txgroup)
+	require.Error(t, err)
+	// Because they fail TestTransactionGroup they won't get added to the txn pool and so they won't actually be run; actually running the txn doesn't check txn.WellFormed()
+	// err = eval.Transaction(st, transactions.ApplyData{})
+	// require.Error(t, err)
```

Comment:
Can you please explain this? I am not able to understand it.
Also, can you please make the line shorter?

Reply:
reworded comment. hopefully clearer.